src/Components/NavBar/NavBar.tsx

```diff
@@ -102,10 +125,24 @@ export const NavBar: React.FC = track({
                     })
                   }}
                 >
-                  <BellIcon top={3} />
+                  {({ hover }) => {
+                    return (
+                      <BellIcon
+                        top={3}
+                        fill={hover ? "purple100" : "black80"}
+                      />
+                    )
+                  }}
```

Comment:
We needed a way to pipe UI states into NavItem contents. For example, most nav items are just text so it's easy to change the text color on hover. But occasionally we need to use icons, which require a prop. Using the renderProps pattern we can pipe internal `NavItem` state up into the upper scope to act on changes.

![nav](https://user-images.githubusercontent.com/236943/58066343-caf84300-7b3d-11e9-9027-e30c46d785a7.gif)